roles/translator/src/lib/proxy_config.rs

```diff
@@ -15,6 +15,42 @@ pub struct ProxyConfig {
     pub upstream_difficulty_config: UpstreamDifficultyConfig,
 }
 
+pub struct TranslatorProxyUpstream {
+    address: String,
+    port: u16,
+    authority_pubkey: Secp256k1PublicKey,
+    difficulty_config: UpstreamDifficultyConfig,
+}
+
+pub struct TranslatorProxyDownstream {
+    address: String,
+    port: u16,
+    difficulty_config: DownstreamDifficultyConfig,
+}
+
+impl ProxyConfig {
+    pub fn new(
```

Comment:
It feels like there could be additional limitation to the extranonce and version fields here. Definitely outside the scope of this PR but worth mentioning for discussion.